### ISSUE TITLE
"No PayPal order found in the current WooCommerce session" error for guests on Pay for Order page (634)

### ISSUE THREAD

**Issue by polkemmet** [woocommerce/woocommerce-paypal-payments#605]
### Describe the Bug

When you manually create a WooCommerce order for a guest, with an email address, and the user attempts to pay via the link in the email, the payment goes through on WooCommerce order, but is marked as failed and the user recieves the above response on the WooCommerce webpage

### Create a manual order, and email the order to customer
Customer uses link to pay 
Payment recieved in PayPal, but error generated on webpage and order marked as failed in WooCommerce Orders screen

1. Go to '…'
2. Click on '…'
3. Scroll down to …'

### Expected Behavior

order should be marked as paid and moved to processing 
Customer gets payment confirmed screen

### Actual Behavior

PayPal payment is actually processed but WooCommerce moves the order to failed. 

### Environment

 - WordPress Version 5.9.3
 - WooCommerce Version 6.4.1
 - Plugin Version1.7.1
 - Browser [e.g. Chrome, Safari] and Version Edge v100
 - Any other plugins installed EverAccounting v1.1.2

### Additional Details

<details><summary>System status</summary>

  ```
  <!--If applicable, paste the system status here. Please ensure you redact or remove any identifying information. -->
  ```

</details>

**Comment by panilo** [comment]
I'm experiencing the same issue manual orders lead to the error “No PayPal order found in the current WooCommerce session”

- WordPress Version 5.9.3
- WooCommerce Version 6.3.1
- Plugin Version1.7.1
- Browser [e.g. Chrome, Safari] and Version Edge v100
- Plugins installed: different plugins are installed 

I disabled all the caches and I’m running my wp on an Apache server. PHP 7.4

**Comment by panilo** [comment]
@InpsydeNiklas would you take a look at this and label it as big? Do you need any further information?

Thanks

**Comment by InpsydeNiklas** [comment]
This issue is being worked on, and we expect a fix in update 1.8.1.
The temporary workaround is to ensure the buyer is logged into a WordPress account, as the problematic behavior only affects guest buyers.

**Comment by kwisatz** [comment]
For an official WooCommerce plug-in, this bug is highly critical in my opinion. Shop owners cannot expect their users to create accounts. In facts most probably don't. When it 1.8.1 estimated to arrive?

**Comment by lakharadk** [comment]
I'm experiencing the same issue on the checkout page the error “No PayPal order found in the current WooCommerce session” and not being able to place an order as the logged-in user. Any ETA for fixing this issue?

**Comment by kxeilakos** [comment]
Does this issue come up on versions prior to 1.8.0 ? Users with version 1.7.0 seem to successfully complete orders for guest users

**Comment by panilo** [comment]
> Does this issue come up on versions prior to 1.8.0 ? Users with version 1.7.0 seem to successfully complete orders for guest users

Version 1.7.1 is affected… have you tested manual orders with checkout as guest?

**Comment by kxeilakos** [comment]
I just tried 1.7.0 and it fails again for guests.

**Comment by nirkons** [comment]
Also facing this exact issue, awaiting eagerly for the fix

**Comment by kxeilakos** [comment]
I tried 1.7.1 no luck again... will give it a try to fix by myself hope it will work

**Comment by mayocube** [comment]
**Background:** 
I studied the plugin and found that the issue is happening because Woocommerce doesn't start session on guest as it requires userID to start a session, the session is used in the code to maintain the order information so that's why it all fails.

**Fix:**
add this code to functions.php of your theme:
```
add_action( 'woocommerce_init', function(){
    if ( WC()->session && !WC()->session->has_session() ) {
        WC()->session->set_customer_session_cookie( true );
    }
} );
```
this will start the session no matter if the person is logged in or not (will just need cookies to be enabled)

**Comment by lakharadk** [comment]
@mayocube 

I just tried your code in child theme functions.php file and got this error 

![image](https://user-images.githubusercontent.com/37809646/167270163-28413ecf-c6d7-4857-aaf9-b8e371490458.png)

Am I doing anything wrong here and is your code tested ?

Thanks

**Comment by mayocube** [comment]
Its tested and it works, do you have **woo commerce** plugin active?

**Comment by mayocube** [comment]
> @mayocube
> 
> I just tried your code in child theme functions.php file and got this error
> 
> ![image](https://user-images.githubusercontent.com/37809646/167270163-28413ecf-c6d7-4857-aaf9-b8e371490458.png)
> 
> Am I doing anything wrong here and is your code tested ?
> 
> Thanks


Its tested and it works, do you have **woo commerce** plugin active?

**Comment by nirkons** [comment]
> > @mayocube
> > I just tried your code in child theme functions.php file and got this error
> > ![image](https://user-images.githubusercontent.com/37809646/167270163-28413ecf-c6d7-4857-aaf9-b8e371490458.png)
> > Am I doing anything wrong here and is your code tested ?
> > Thanks
> 
> Its tested and it works, do you have **woo commerce** plugin active?

For me also,
I think checkout process works properly now but when i try to log in to wp-admin i get this
```
` Fatal error: Uncaught Error: Call to a member function has_session() on null
in /var/www/html/app/wp-content/themes/hello-elementor-child/functions.php on line 349

Call stack:

{closure}()
wp-includes/class-wp-hook.php:307
WP_Hook::apply_filters()
wp-includes/class-wp-hook.php:331
WP_Hook::do_action()
wp-includes/plugin.php:474
do_action()
wp-content/plugins/woocommerce/includes/class-woocommerce.php:635
WooCommerce::init()
wp-includes/class-wp-hook.php:307
WP_Hook::apply_filters()
wp-includes/class-wp-hook.php:331
WP_Hook::do_action()
wp-includes/plugin.php:474
do_action()
wp-settings.php:587
require_once()
wp-config.php:93
require_once()
wp-load.php:50
require_once()
wp-admin/admin.php:34
require_once()
wp-admin/index.php:10`
```

**Comment by mayocube** [comment]
@nirkons @lakharadk 

Here you go this should fix your error.
```
if ( WC()->session && !WC()->session->has_session() ) {
     WC()->session->set_customer_session_cookie( true );
}
```

**Comment by nirkons** [comment]
> @nirkons @lakharadk
> 
> Here you go this should fix your error.
> 
> ```
> if ( WC()->session && !WC()->session->has_session() ) {
>      WC()->session->set_customer_session_cookie( true );
> }
> ```

thanks but I don't understand, isn't that the same exact code as before?

**Comment by nirkons** [comment]
> @nirkons @lakharadk
> 
> Here you go this should fix your error.
> 
> ```
> if ( WC()->session && !WC()->session->has_session() ) {
>      WC()->session->set_customer_session_cookie( true );
> }
> ```

I think its working :)
thanks!

**Comment by lakharadk** [comment]
@mayocube @nirkons 

This worked. Thanks.

**Comment by polkemmet** [comment]
> @nirkons @lakharadk
> 
> Here you go this should fix your error.
> 
> ```
> if ( WC()->session && !WC()->session->has_session() ) {
>      WC()->session->set_customer_session_cookie( true );
> }
> ```

Applied using Code Snippet and working successfully

**Comment by MikeSamy** [comment]
> This issue is being worked on, and we expect a fix in update 1.8.1. The temporary workaround is to ensure the buyer is logged into a WordPress account, as the problematic behavior only affects guest buyers.

Our site only allows users to check out if logged in, but we still experience this issue intermittently. Running 1.8.0.

**Comment by Dinamiko** [comment]
Fixed in #641, will be included in 1.9.0 release.
